tests/ptvsd/server/test_attach.py

```diff
@@ -6,55 +6,47 @@
 
 import pytest
 
-from tests import code, debug, test_data
+from tests import debug, test_data
 from tests.patterns import some
 from tests.timeline import Event
 
 
-@pytest.mark.parametrize("wait_for_attach", ["waitOn", "waitOff"])
-@pytest.mark.parametrize("is_attached", ["attachCheckOn", "attachCheckOff"])
-@pytest.mark.parametrize("break_into", ["break", "pause"])
-def test_attach(run_as, wait_for_attach, is_attached, break_into):
+@pytest.mark.parametrize("wait_for_attach", ["wait_for_attach", ""])
+@pytest.mark.parametrize("is_attached", ["is_attached", ""])
+@pytest.mark.parametrize("break_into_debugger", ["break_into_debugger", ""])
+def test_attach(run_as, wait_for_attach, is_attached, break_into_debugger):
     attach1_py = test_data / "attach" / "attach1.py"
-    lines = code.get_marked_line_numbers(attach1_py)
-    with debug.Session() as session:
-        env = {
-            "PTVSD_TEST_HOST": "localhost",
-            "PTVSD_TEST_PORT": str(session.ptvsd_port),
-        }
-        if wait_for_attach == "waitOn":
-            env["PTVSD_WAIT_FOR_ATTACH"] = "1"
-        if is_attached == "attachCheckOn":
-            env["PTVSD_IS_ATTACHED"] = "1"
-        if break_into == "break":
-            env["PTVSD_BREAK_INTO_DBG"] = "1"
+
+    with debug.Session("custom_server") as session:
+        session.env.update({
+            "ATTACH1_TEST_PORT": str(session.ptvsd_port),
+            "ATTACH1_WAIT_FOR_ATTACH": "1" if wait_for_attach else "0",
+            "ATTACH1_IS_ATTACHED":  "1" if is_attached else "0",
+            "ATTACH1_BREAK_INTO_DEBUGGER": "1" if break_into_debugger else "0",
+        })
 
         backchannel = session.setup_backchannel()
-        session.initialize(
-            target=(run_as, attach1_py),
-            start_method="launch",
-            env=env,
-        )
+        session.initialize(target=(run_as, attach1_py))
         session.start_debugging()
 
-        if wait_for_attach == "waitOn":
+        if wait_for_attach:
```

Comment:
do we need bool() for clarity?

Reply:
I think it's a common enough pattern ("" vs "foo") in tests that it's not worth the bother. Since there's no chance of `None` here, the usual problem with distinguishing it vs empty string is not applicable.